### PR TITLE
Italian-Docs

### PR DESCRIPTION
**Pull request by justoverclockl**
Italian translation for flarum docs.

### CODE REVIEW

**Review by askvortsov1 — CHANGES_REQUESTED** [review]
Thank you for the PR! These should be added as separate files with a navigation configuration similar to that of the chinese and turkish translations we have now.

**Comment by justoverclockl** [comment]
can u explain me step by step what i have to do? i'm really new to github :)

u mean that i have to create a folder [it]?

**Comment by Littlegolden** [comment]
@justoverclockl I hope this tutorial can help you~ https://discuss.flarum.org.cn/d/2433

**Comment by justoverclockl** [comment]
> @justoverclockl I hope this tutorial can help you~ https://discuss.flarum.org.cn/d/2433

mmmhhh not really, i need to install vuepress locally on my computer?

**Comment by Littlegolden** [comment]
If you don't need to preview the docs website locally, you don't need to run `npm install`, just config the locale settings and create the folder where the Italian documentation will be stored.

**Comment by justoverclockl** [comment]
> If you don't need to preview the docs website locally, you don't need to run `npm install`, just config the locale settings and create the folder where the Italian documentation will be stored.

can u check now, now the folder is updated with original structure, and IT folder with translation

**Comment by Littlegolden** [comment]
yes it's right now. and don't forget the config files in `/.vuepress/config`

**Comment by justoverclockl** [comment]
> yes it's right now. and don't forget the config files in `/.vuepress/config`

done!

File: docs/.vuepress/config/locales/it/nav.js
Comment (Littlegolden) [inline docs/.vuepress/config/locales/it/nav.js:2]: the link shoule be `/it/`

File: docs/.vuepress/config/locales/it/nav.js
Comment (Littlegolden) [inline docs/.vuepress/config/locales/it/nav.js:3]: you can keep extend docs link `/extent` before you translated extend docs, this will link readers to the English document

File: docs/.vuepress/config/locales/it/sidebar.js
Comment (Littlegolden) [inline docs/.vuepress/config/locales/it/sidebar.js:68]: need change to `/it/`

File: docs/.vuepress/config/locales/it/nav.js
Comment (justoverclockl) [reply docs/.vuepress/config/locales/it/nav.js]: done!

File: docs/.vuepress/config/locales/it/sidebar.js
Comment (justoverclockl) [reply docs/.vuepress/config/locales/it/sidebar.js]: done!

**Review by askvortsov1 — CHANGES_REQUESTED** [review]
Overall, looking quite good! A few more changes needed:

- `'it'` needs to be added to `.vuepress/index.js`'s `lang` array so that Italian shows up in the top right "select language" dropdown
- For extend docs that haven't yet been translated, instead of copy-pasting the English version, you should put a short message saying that they haven't been translated, and to refer to the English version. We want to avoid maintaining multiple copies of translations in the same language so that  changes can be made in one place (for that language)

**Comment by justoverclockl** [comment]
> Overall, looking quite good! A few more changes needed:
> 
> * `'it'` needs to be added to `.vuepress/index.js`'s `lang` array so that Italian shows up in the top right "select language" dropdown
> * For extend docs that haven't yet been translated, instead of copy-pasting the English version, you should put a short message saying that they haven't been translated, and to refer to the English version. We want to avoid maintaining multiple copies of translations in the same language so that  changes can be made in one place (for that language)

vuepress/index.js added.

i'm new to github so i do not understand well what i have to do with the Extend folder (actually i'm working on it locally)

File: docs/.vuepress/config/locales/index.js
Comment (askvortsov1) [inline docs/.vuepress/config/locales/index.js:2]: This should be restored to how it was before

**Review by askvortsov1 — CHANGES_REQUESTED** [review]
Ah, sorry I see now that you're working through translations for the extend stuff; that's fine then, we'll just wait till they're done to merge this.

One thing we'll want to add: since the dev team writes docs in English, translated docs might get out of date. We should add a notice about this similar to how the Chinese docs do (https://docs.flarum.org/zh/, for instance)

File: docs/.vuepress/config/locales/index.js
Comment (Littlegolden) [reply docs/.vuepress/config/locales/index.js:2]: In fact, the path does not need to be modified

Comment (askvortsov1) [reply docs/.vuepress/config/locales/index.js:2]: `langs` should stay as it is now (with the `it` added), but `paths` should be changed back; this configures it so that the english docs will show up at `docs.flarum.org` instead of `docs.flarum.org/en` since those are the default

Comment (justoverclockl) [reply docs/.vuepress/config/locales/index.js]: oh, sorry for that, done!

**Comment by justoverclockl** [comment]
> Overall, looking quite good! A few more changes needed:
> 
> * `'it'` needs to be added to `.vuepress/index.js`'s `lang` array so that Italian shows up in the top right "select language" dropdown
> * For extend docs that haven't yet been translated, instead of copy-pasting the English version, you should put a short message saying that they haven't been translated, and to refer to the English version. We want to avoid maintaining multiple copies of translations in the same language so that  changes can be made in one place (for that language)

ok, can u tell me where this need to be applied? and maybe an example of the text? on that links i see only chinese words :)

**Comment by Littlegolden** [comment]
The notice component is [here](https://github.com/flarum/docs/tree/master/docs/.vuepress/components)，take a look at how the Chinese docs use it. https://github.com/flarum/docs/blame/master/docs/zh/install.md#L1-L3

**Comment by justoverclockl** [comment]
> The notice component is [here](https://github.com/flarum/docs/tree/master/docs/.vuepress/components)，take a look at how the Chinese docs use it: https://github.com/flarum/docs/blame/master/docs/zh/install.md#L1-L3
> 
> You can just use the component for now. I believe we can use js to insert notice text in different languages which I can do that in the daytime tomorrow.

done (not sure, but done) :)

**Comment by askvortsov1** [comment]
> > The notice component is [here](https://github.com/flarum/docs/tree/master/docs/.vuepress/components)，take a look at how the Chinese docs use it: https://github.com/flarum/docs/blame/master/docs/zh/install.md#L1-L3
> > You can just use the component for now. I believe we can use js to insert notice text in different languages which I can do that in the daytime tomorrow.
> 
> done (not sure, but done) :)

Not quite. You need to copy that file (outdated from components) to a new one with `it` in the name, replace the Chinese text with italian equivalent, and reference that new component at the top of every time

File: docs/it/README.md
Comment (matteocontrini) [inline docs/it/README.md:15]:
```suggestion
* **Potente ed estendibile.** Personalizza, estendi e integra Flarum per creare la tua community unica. L'architettura di Flarum è molto flessibile grazie alle sue [potenti API per le estensioni](/extend/).
```

File: docs/it/README.md
Comment (matteocontrini) [inline docs/it/README.md:17]:
```suggestion
* **Gratuito e aperto.** Flarum è totalmente gratuito e rilasciato sotto [licenza MIT](https://github.com/flarum/flarum/blob/master/LICENSE).
```

File: docs/it/admin.md
Comment (matteocontrini) [inline docs/it/admin.md:4]:
```suggestion
È disponibile solo per gli utenti nel [gruppo](permissions.md) "Admin".
```

File: docs/it/bugs.md
Comment (matteocontrini) [inline docs/it/bugs.md:13]:
```suggestion
- Cerca sul [forum di supporto](https://discuss.flarum.org/t/support) per vedere se il bug è stato già segnalato.
```

File: docs/it/bugs.md
Comment (matteocontrini) [inline docs/it/bugs.md:14]:
```suggestion
- Potremmo essere già al lavoro su una soluzione, quindi cerca anche sul nostro [sistema di gestione degli *issue*](https://github.com/flarum/core/issues).
```

File: docs/it/bugs.md
Comment (matteocontrini) [inline docs/it/bugs.md:16]:
```suggestion
Se hai cercato a fondo e ne sei uscito a mani vuote, saremo lieti di ricevere il tuo rapporto. Se si tratta solo di un problema semplice (ad esempio una parola con errori di ortografia o un problema grafico), passa al paragrafo successivo. Ma se vedi errori o qualcosa è chiaramente un errore, avremo bisogno che tu raccolga prima alcune informazioni. Per favore leggi la guida per la [Risoluzione dei problemi](troubleshoot.md)  e seguine le istruzioni. Raccogli più informazioni possibili!
```

File: docs/it/code-of-conduct.md
Comment (matteocontrini) [inline docs/it/code-of-conduct.md:25]:
```suggestion
Vuoi iniziare una nuova discussione? Innanzitutto, assicurati di leggere le nostre [domande frequenti](faq.md) ae segui i link per assicurarti di essere completamente informato sul progetto. Quindi trascorri un po 'di tempo a navigare nel forum, familiarizza con [il sistema dei TAG](https://discuss.flarum.org/tags), e fai qualche ricerca per parole chiave correlate al tuo argomento: ci potrebbe essere qualcuno che ha già avviato una discussione simile o identica!
```

File: docs/it/code-of-conduct.md
Comment (matteocontrini) [inline docs/it/code-of-conduct.md:35]: Missing translation.

File: docs/it/config.md
Comment (matteocontrini) [inline docs/it/config.md:28]:
```suggestion
  'url' => 'https://flarum.localhost', // l'URL di installazione, vorrai cambiarlo se cambi domini
```

File: docs/it/contributing.md
Comment (matteocontrini) [inline docs/it/contributing.md:11]:
```suggestion
Dai un occhiata ai nostri prossimi [Traguardi (in inglese)](https://github.com/flarum/core/milestones) per una panoramica di ciò che deve essere fatto. Consulta le [Good first issue](https://github.com/flarum/core/labels/Good%20first%20issue) per un elenco di problemi con cui dovrebbe essere relativamente facile iniziare.
```

**Review by matteocontrini — CHANGES_REQUESTED** [review]
Hello, I left some changes suggestions. It seems there are many other links that are untranslated, I just skimmed quickly through the first few files, but unfortunately I don't have time to review the whole thing, it's extremely long 😰

**Comment by justoverclockl** [comment]
> Hello, I left some changes suggestions. It seems there are many other links that are untranslated, I just skimmed quickly through the first few files, but unfortunately I don't have time to review the whole thing, it's extremely long 😰

I will check again all to fix this asap!

**Comment by luceos** [comment]
@matteocontrini and @Littlegolden thanks for reviewing, if the both of you can approve once everything is in order I think it's okay to merge 👍

**Review by matteocontrini — APPROVED** [review]
I'm unable to read all of it thoroughly, but I skimmed through it and pushed some fixes, especially with accents that were broken in some files. I also changed the wording of some technical terms in the Contributing page, so that the original English terms are used (e.g. "branch" shouldn't be translated).

Overall fine for me :)

**Comment by justoverclockl** [comment]
> I'm unable to read all of it thoroughly, but I skimmed through it and pushed some fixes, especially with accents that were broken in some files. I also changed the wording of some technical terms in the Contributing page, so that the original English terms are used (e.g. "branch" shouldn't be translated).
> 
> Overall fine for me :)

we waiting for @luceos

File: docs/.vuepress/config/locales/it/nav.js
Comment (askvortsov1) [inline docs/.vuepress/config/locales/it/nav.js:2]: Should these be translated?

Comment (askvortsov1) [reply docs/.vuepress/config/locales/it/nav.js:2]: The link texts I mean

File: docs/.vuepress/config/locales/it/nav.js
Comment (askvortsov1) [inline docs/.vuepress/config/locales/it/nav.js:3]: Since extend docs are now translated, this should be switched back to '/it/extend/'

File: docs/.vuepress/config/locales/it/sidebar.js
Comment (askvortsov1) [inline docs/.vuepress/config/locales/it/sidebar.js:2]: I think this needs to be '/it/extend/'

File: docs/it/code-of-conduct.md
Comment (askvortsov1) [inline docs/it/code-of-conduct.md:35]: This is still missing

File: docs/it/extend/forms.md
Comment (askvortsov1) [inline docs/it/extend/forms.md:1]: For pages under extend that aren't yet translated, please don't include the English source; instead, leave it blank and add text saying that translation is still in progress.

**Review by askvortsov1 — CHANGES_REQUESTED** [review]
Found a few more things. Also, these pages still need the "this might get slightly outdated" disclaimer

File: docs/.vuepress/config/locales/it/nav.js
Comment (justoverclockl) [reply docs/.vuepress/config/locales/it/nav.js:2]: link is usable even in italian

**Comment by justoverclockl** [comment]
> @matteocontrini and @Littlegolden thanks for reviewing, if the both of you can approve once everything is in order I think it's okay to merge 👍

for the component "outdated", i'm really confusing, i think i need some help

**Comment by askvortsov1** [comment]
I've unresolved a few more comments above that still need to be addressed.

For the "outdated" component, take a look at how this is implemented in the Chinese docs. If you're not comfortable setting that up, that's alright, we can add it in separately after merging the PR.

**Comment by justoverclockl** [comment]
> I've unresolved a few more comments above that still need to be addressed.
> 
> For the "outdated" component, take a look at how this is implemented in the Chinese docs. If you're not comfortable setting that up, that's alright, we can add it in separately after merging the PR.

i think coul be merged now. all translation are ok except for the "outdated" componet, not comfortable with this

File: docs/.vuepress/config/locales/it/nav.js
Comment (askvortsov1) [inline docs/.vuepress/config/locales/it/nav.js:3]: I'm pretty sure this needs to be `/it/extend`

File: docs/it/extend/update-b10.md
Comment (askvortsov1) [inline docs/it/extend/update-b10.md:1]: If these aren't gonna be translated (which is fine, they're outdated and not essential) they should be removed.

**Review by askvortsov1 — CHANGES_REQUESTED** [review]
Thank you so much for the collossal work done on this! A few last comments:

- I think I found one last broken link
- The upgrade guides can probably be completely removed

**Comment by justoverclockl** [comment]
> Thank you so much for the collossal work done on this! A few last comments:
> 
> * I think I found one last broken link
> * The upgrade guides can probably be completely removed

its a pleasere, i really love your works, thats the minimum i can do to give you a little help

anyway i've removed (i hope) the useless file, and added correct path on nav.js

File: .gitignore
Comment (askvortsov1) [inline .gitignore:5]: These (gitignore) changes are unnecessary; also, links to the upgrade guides need to be removed from the navbar

**Review by askvortsov1 — CHANGES_REQUESTED** [review]

File: .gitignore
Comment (justoverclockl) [reply .gitignore:5]: how can i delete this file? (i'm new to github)

Comment (askvortsov1) [reply .gitignore:5]: No need to delete the files, just remove the added lines referencing .md files.

Comment (justoverclockl) [reply .gitignore:5]: i think it's done!

**Comment by askvortsov1** [comment]
One last thing: could you translate the following text to italian?

> These documentation translations are community maintained, and might be outdated at times. Please refer to the English documentation for the latest version. If there are issues, please let us know [on github](https://github.com/flarum/docs/issues).

**Comment by justoverclockl** [comment]
> One last thing: could you translate the following text to italian?
> 
> > These documentation translations are community maintained, and might be outdated at times. Please refer to the English documentation for the latest version. If there are issues, please let us know [on github](https://github.com/flarum/docs/issues).

Queste traduzioni della documentazione sono gestite dalla comunità e potrebbero essere obsolete. Fare riferimento alla documentazione in inglese per la versione più recente. Se riscontrate problemi, vi consigliamo di scriverci su GitHub.

**Review by askvortsov1 — APPROVED** [review]

**Comment by justoverclockl** [comment]
> Thank you again so much for this collosal effort!

is the minimum that i can do, i will make another check for few little errors

**Pull request merged**
